### PR TITLE
Implement Apple Rosetta translation check

### PR DESCRIPTION
**Pull request by GediminasZukas**
Solves: https://github.com/runningcode/gradle-doctor/issues/209

Used `sysctl.proc_translated` API via bash in order to determine if process is running under Rosetta or not according to documentation: https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment#Determine-Whether-Your-App-Is-Running-as-a-Translated-Binary

Additional related thought: maybe we need to make this check more generic and compare machine CPU architecture with running JDK target architecture? Something similar to: https://github.com/runningcode/gradle-doctor/issues/209#issuecomment-1132645634  As hardware for Windows or Linux may benefit from ARM architecture in the future as well (maybe already there is some variants in the market).

### CODE REVIEW

**Review by runningcode — CHANGES_REQUESTED** [review]
thanks for doing this! just a comment on the commandline usage

File: doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.osacky.doctor.internal
+
+import java.io.InputStream
+
+class CliCommandExecutor {
```
[the remark this answers is not in the document]
Comment (GediminasZukas) [reply doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt]: Made a change. Additionally had to bump Gradle to the newest 7.5 version in order to use new API. Thus, check won't be backwards compatible and will work only if clients will use Gradle 7.5+.

[the remark this answers is not in the document]
Comment (GediminasZukas) [reply doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt]: Maybe better to update Gradle in a separate PR?

[the remark this answers is not in the document]
Comment (runningcode) [reply doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt]: Ah good point. Let's see which tests pass here. There is a cross version compatibility test which might not work.

**Comment by runningcode** [comment]
Thanks for the ping. Running tests now.

File: doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (GediminasZukas) [reply doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt]: Tests run failed due to lint errors. Fixed it with most recent commit.

File: doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt [hunk 2]
```diff
@@ -26,7 +26,7 @@ class PluginIntegrationTest constructor(private val version: String) {
         fun getParams(): List<String> {
             // Keep 6.0 as minimum unsupported version and 6.1 as minimum supported version.
             // Keep this list to 5 as testing against too many versions causes OOMs.
```
Comment (julioz) [inline doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt:28]: with the change, this comment rotted: there are now 6 versions in the returned list, going against the advice here.

File: doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (GediminasZukas) [reply doctor-plugin/src/main/java/com/osacky/doctor/internal/CliCommandExecutor.kt]: @runningcode, could you review again and launch tests? Seems found the cause of failing test and fixed it.

File: doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt (hunk 2 above)
Comment (GediminasZukas) [reply doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt:28]: Good question, actually we have 7 versions here now. @runningcode, how to proceed with this?

**Comment by GediminasZukas** [comment]
@runningcode, friendly ping. How we can proceed further with this change?

File: doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt (hunk 2 above)
Comment (runningcode) [reply doctor-plugin/src/test/java/com/osacky/doctor/PluginIntegrationTest.kt:28]: let's remove 6.1.1 and 7.3.3

**Review by runningcode — APPROVED** [review]
thanks for the ping, please remove those two versions to keep the list to 5 and we can proceed!

**Comment by GediminasZukas** [comment]
Done!

**Comment by GediminasZukas** [comment]
@runningcode, can we merge this?

**Pull request merged**